src/Symfony/Component/Translation/MessageSelector.php

```diff
@@ -47,11 +47,11 @@ class MessageSelector
      */
     public function choose($message, $number, $locale)
     {
-        $parts = explode('|', $message);
+        $parts = preg_split('/(?<![^\\\\]\\\\)\|/', $message);
```

Comment:
It doesn't work as expected with "a\\\\\|b" (escaped backslash, escaped backslash, escaped "|"). I don't think it's possible to solve it properly with single regex.